libs/auth-nest-tools/src/lib/header.middleware.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * We proxy the auth header to the subsystem where it is resolved.
+ */
+interface FetchParams {
+  url: string
+  init: RequestInit
+}
+
+interface RequestContext {
+  init: RequestInit
+}
+
+interface Middleware {
+  pre?(context: RequestContext): Promise<FetchParams | void>
+}
+
+export class HeaderMiddleware implements Middleware {
```

Comment:
IMO `HeaderMiddleware` sounds like you can override any header. I suggested `AuthHeaderMiddleware` before. 👍